### PR TITLE
Adds various new conventions

### PR DESCRIPTION
**Pull request by flakey-bit**
This PR adds various new conventions (discussed in more detail in #87)

Things to note:
* I've not added `NamespaceMustStartWith` because it turns out something equivalent already existed (not sure how I missed it)
* I've opened an issue #88 which affects at least one existing convention but also affects most of the conventions this PR introduces
* To support legacy csproj files, the code uses an XPath that explicitly looks only at the local name of elements, however let me know if you'd prefer a different approach.

### CODE REVIEW

File: src/Core/Conventional.Tests/AllAssembliesScenarios.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ public void GivenAPattern_LocatesAndReturnsAllAssembliesForThatPattern()
         {
             var assemblySpecimen = AllAssemblies.WithNamesMatching("*");
 
-            assemblySpecimen.Should().HaveCount(5);
+            assemblySpecimen.Should().HaveCount(6);
```
Comment (flakey-bit) [inline src/Core/Conventional.Tests/AllAssembliesScenarios.cs:13]: **remark**: New SDK project introduced

File: src/Core/TestSolution/TestSolution.TestProject/SdkClassLibrary1/SdkClassLibrary1.csproj [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    
+    <PropertyGroup>
+        <WarningsAsErrors>CS0162</WarningsAsErrors>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <PackageReference Include="coverlet.collector" Version="6.0.0">
+        <PrivateAssets>all</PrivateAssets>
+        <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      </PackageReference>
```
Comment (flakey-bit) [inline src/Core/TestSolution/TestSolution.TestProject/SdkClassLibrary1/SdkClassLibrary1.csproj:21]: **remark**: Let me know if you're unhappy with this package reference (in the test project) for some reason.

File: src/Core/Conventional.Tests/Conventional/Conventions/Cecil/CecilConventionSpecificationTests.cs [hunk 3]
```diff
@@ -515,5 +515,63 @@ public void LibraryCodeShouldCallConfigureAwaitWhenAwaitingTasks_FailsWhenConfig
             result.IsSatisfied.Should().BeFalse();
             result.Failures.Single().Should().Be(expectedFailureMessage);
         }
+
+        # region MustNotUseGuidNewGuid
```
Comment (flakey-bit) [inline src/Core/Conventional.Tests/Conventional/Conventions/Cecil/CecilConventionSpecificationTests.cs:519]: **remark**: I've introduced a region here to make it obvious that all of this is associated with the `MustNotUseGuidNewGuid` convention. 

Might be worth splitting this file up.

Comment (andrewabest) [reply src/Core/Conventional.Tests/Conventional/Conventions/Cecil/CecilConventionSpecificationTests.cs:519]: I'm not too worried about splitting these test files up, they tend to be easy enough to peruse. Can we remove the `#region` though? It conflicts with my core beliefs 😆

File: src/Core/TestSolution/TestSolution.TestProject/SdkClassLibrary1/SdkClassLibrary1.csproj (hunk 2 above)
Comment (andrewabest) [reply src/Core/TestSolution/TestSolution.TestProject/SdkClassLibrary1/SdkClassLibrary1.csproj:21]: This is fine 👍

**Review by andrewabest — APPROVED** [review]
An excellent changeset 👌 I don't have any feedback to address on this one. The implementations look good, the test coverage looks good, and the code follows the codebases conventions... 😉 Thankyou for the contribution!

**Comment by andrewabest** [comment]
Fixes #87

**Comment by andrewabest** [comment]
Ah... just forgot about that sneaky region before hitting merge! I'll remove it @flakey-bit.

**Pull request merged**